src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +119,343 @@ def execute(self, reqStatus):
+        """
+        A function intended to dispatch a workflow (e.g based on its status)
+        through one or more functional pipelines in case there is some more
+        complicated logic involved in the order we execute them but not just
+        a sequentially
+        """
+        # msg = "Called with wflow: %s"
+        # self.logger.debug(msg, pformat(wflow))
+        # NOTE: The following dispatch logic is a subject to be changed at any time
+
+        # Resolve:
+        # NOTE: First resolve any preliminary flags that will be needed further
+        #       in the logic of the _dispatcher() itself
+        if wflow['RequestStatus'] == 'announced':
+            self.getMSOutputTransferInfo(wflow)
+
+        # Clean:
+        # Do not clean any Resubmission, but still let them be archived
+        if wflow['RequestType'] == 'Resubmission':
+            wflow['ForceArchive'] = True
+            msg = "Skipping cleanup step for workflow: %s - RequestType is %s."
+            msg += " Will try to archive it directly."
+            self.logger.info(msg, wflow['RequestName'], wflow['RequestType'])
+        elif wflow['RequestStatus'] in ['rejected', 'aborted-completed']:
+            # NOTE: We do not check the ParentageResolved flag for these
+            #       workflows, but we do need to clean output data placement
+            #       rules from the agents for them
+            for pline in self.agentlines:
+                try:
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline. Workflow: %s. Error: \n%s. "
+                    msg += "\nWill retry again in the next cycle."
+                    self.logger.exception(msg, pline.name, wflow['RequestName'], str(ex))
+                    continue
+                if wflow['CleanupStatus'][pline.name]:
+                    self.wfCounters['cleaned'][pline.name] += 1
+        elif wflow['RequestStatus'] == 'announced' and not wflow['ParentageResolved']:
+            # NOTE: We skip workflows which are not having 'ParentageResolved'
+            #       flag, but we still need some proper logging for them.
+            msg = "Skipping workflow: %s - 'ParentageResolved' flag set to false."
+            msg += " Will retry again in the next cycle."
+            self.logger.info(msg, wflow['RequestName'])
+        elif wflow['RequestStatus'] == 'announced' and not wflow['TransferDone']:
+            # NOTE: We skip workflows which have not yet finalised their TransferStatus
+            #       in MSOutput, but we still need some proper logging for them.
+            msg = "Skipping workflow: %s - 'TransferStatus' is 'pending' or 'TransferInfo' is missing in MSOutput."
+            msg += " Will retry again in the next cycle."
+            self.logger.info(msg, wflow['RequestName'])
+        elif wflow['RequestStatus'] == 'announced' and wflow['ParentageResolved'] and wflow['TransferDone']:
```

Comment:
Only this:
```
elif wflow['RequestStatus'] == 'announced':
```
would do!

Reply:
Allow me to disagree on this one. The very strict definition on which workflows to work in this line is by purpose. And it is because if we generalize the condition here, alot of things that have not been thought on would pass under the generalized condition, and in case there is an error in the logic which workflows reach to here (as an example someone decides to tamper with or by mistake touches the list of statuses for the component run) then all those will pass through unnoticed and unpredictable behavior is guaranteed. But on the other hand if we see something falling to the `else` statement a line bellow this is going to be a sure signal of something wrong, wile protecting us from screwing a workflow.

Reply:
What I am trying to say is basically this one is there to protect ourselves from our won mistakes, and for me it makes it much clearer and visibe which is the full set of conditions about what workflow Ii expect here.

Reply:
And those would still be protected with the suggestion I made. Announced workflows have to satisfy only 2 conditions:
* ParentageResolved
* TransferDone
which are already evaluated in the previous `elif`. I'm also not suggesting you to remove the `else`.
My suggetion makes the code cleaner and avoid repetitive/duplicate check conditions.

Reply:
You have it.